src/Castle.Core.AsyncInterceptor/IAsyncActionInvocation.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) 2016 James Skimming. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace Castle.DynamicProxy
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Interface to handle synchronous function
+    /// </summary>
+    public interface IAsyncActionInvocation : IAsyncInvocation
+    {
+        /// <summary>
+        ///   Proceeds the call to the next interceptor in line, and ultimately to the target method.
+        /// </summary>
+        /// <remarks>
+        ///   Since interface proxies without a target don't have the target implementation to proceed to,
+        ///   it is important, that the last interceptor does not call this method, otherwise a
+        ///   <see cref = "NotImplementedException" /> will be thrown.
+        /// </remarks>
+        /// <returns>The asynchronous task</returns>
+        Task Proceed();
```

Comment:
What about naming ProceedAsync?

Reply:
Personally, I don't like to have the `Async` suffix.

But it is important to express intent and since the synchronous version is called `Proceed` is a good idea to change to `ProceedAsync` to be extra clear.